[BUG] timeout: cog throws error

**Describe the bug**

Timeout cog throws a TypeError from the timeout state conditional logic.

**To Reproduce**
Steps to reproduce the behavior:
1. Run `[p]timeout <user>`
2. See error

```py
[2023-05-10 10:28:51] [ERROR] red: Exception in command 'timeout'
Traceback (most recent call last):
  File "/home/redbot/.local/lib/python3.11/site-packages/discord/ext/commands/core.py", line 229, in wrapped
    ret = await coro(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/redbot/data/cogs/CogManager/cogs/timeout/timeout.py", line 306, in timeout
    if set(user.roles) == {everyone_role} | {timeout_roleset}:
                                            ^^^^^^^^^^^^^^^^^
TypeError: unhashable type: 'set'
```

--- a/timeout/timeout.py
+++ b/timeout/timeout.py
@@ -6,6 +6,7 @@ from redbot.core import Config, checks, commands
 
 log = logging.getLogger("red.rhomelab.timeout")
 
+
 class Timeout(commands.Cog):
     """Timeout a user"""
 
@@ -20,6 +21,9 @@ class Timeout(commands.Cog):
         self.config.register_member(
             roles=[]
         )
+
+        self.actor: str = None
+        self.target: str = None
 
     # Helper functions
 
@@ -59,13 +63,20 @@ class Timeout(commands.Cog):
         # Send embed
         await log_channel.send(embed=embed)
 
-    async def timeout_add(self, ctx: commands.Context, user: discord.Member, reason: str, timeout_role: discord.Role, timeout_roleset: list[discord.Role]):
+    async def timeout_add(
+            self, ctx: commands.Context,
+            user: discord.Member,
+            reason: str,
+            timeout_role: discord.Role,
+            timeout_roleset: list[discord.Role]):
         """Retrieve and save user's roles, then add user to timeout"""
         # Catch users already holding timeout role.
         # This could be caused by an error in this cog's logic or,
         # more likely, someone manually adding the user to the role.
         if timeout_role in user.roles:
-            await ctx.send("Something went wrong! Is the user already in timeout? Please check the console for more information.")
+            await ctx.send(
+                "Something went wrong! Is the user already in timeout? Please check the console for more information."
+            )
             log.warning(
                 f"Something went wrong while trying to add user {self.target} to timeout.\n" +
                 f"Current roles: {user.roles}\n" +
@@ -81,7 +92,7 @@ class Timeout(commands.Cog):
         # Replace all of a user's roles with timeout roleset
         try:
             await user.edit(roles=timeout_roleset)
-            log.info(f"User {self.target} added to timeout by {self.actor}.")
+            log.info("User %s added to timeout by %s.", self.target, self.actor)
         except AttributeError:
             await ctx.send("Please set the timeout role using `[p]timeoutset role`.")
             return
@@ -121,7 +132,7 @@ class Timeout(commands.Cog):
         # Replace user's roles with their previous roles.
         try:
             await user.edit(roles=user_roles)
-            log.info(f"User {self.target} removed from timeout by {self.actor}.")
+            log.info("User %s removed from timeout by %s.", self.target, self.actor)
         except discord.Forbidden as error:
             await ctx.send("Whoops, looks like I don't have permission to do that.")
             log.exception(

--- a/timeout/timeout.py
+++ b/timeout/timeout.py
@@ -103,7 +103,6 @@ class Timeout(commands.Cog):
                 f"Current roles: {user.roles}\n" +
                 f"Attempted new roles: {timeout_roleset}", exc_info=error
             )
-            return
         except discord.HTTPException as error:
             await ctx.send("Something went wrong! Please check the console for more information.")
             log.exception(
@@ -140,7 +139,6 @@ class Timeout(commands.Cog):
                 f"Current roles: {user.roles}\n" +
                 f"Attempted new roles: {user_roles}", exc_info=error
             )
-            return
         except discord.HTTPException as error:
             await ctx.send("Something went wrong! Please check the console for more information.")
             log.exception(
@@ -148,7 +146,6 @@ class Timeout(commands.Cog):
                 f"Current roles: {user.roles}\n" +
                 f"Attempted new roles: {user_roles}", exc_info=error
             )
-            return
         else:
             await ctx.message.add_reaction("✅")
 

--- a/timeout/timeout.py
+++ b/timeout/timeout.py
@@ -303,7 +303,7 @@ class Timeout(commands.Cog):
 
         # Check if user already in timeout.
         # Remove & restore if so, else add to timeout.
-        if set(user.roles) == {everyone_role} | {timeout_roleset}:
+        if set(user.roles) == {everyone_role} | timeout_roleset:
             await self.timeout_remove(ctx, user, reason)
 
         else:

